pom.xml

```diff
@@ -1826,6 +1826,22 @@
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-dependency-plugin</artifactId>
                     <version>3.1.1</version>
+                    <executions>
+                        <execution>
+                            <id>analyze</id>
+                            <goals>
+                                <goal>analyze-only</goal>
+                            </goals>
+                            <configuration>
+                                <failOnWarning>true</failOnWarning>
+                                <!-- ignore annotations for "unused but declared" warnings -->
```

Comment:
These dependencies required at compile time.

We need to ignore warnings:

```
Warning:  Unused declared dependencies found:
Warning:     org.apache.iceberg:iceberg-aws:jar:1.6.1:compile
Warning:     software.amazon.awssdk:glue:jar:2.28.28:compile
```

Reply:
I don't see any compile time usages for these dependencies. Can we try setting `runtime` scope for these dependencies?

Reply:
Actually these dependency is needed at compile time otherwise we will have errors for testcases : 
```
Error:  org.apache.druid.iceberg.input.GlueIcebergCatalogTest.testCatalogCreate -- Time elapsed: 0.001 s <<< ERROR!
java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: software/amazon/awssdk/services/sts/model/Tag
	at org.apache.iceberg.aws.AwsProperties.toStsTags(AwsProperties.java:412)
	at org.apache.iceberg.aws.AwsProperties.<init>(AwsProperties.java:264)
	at org.apache.iceberg.aws.AwsClientFactories$DefaultAwsClientFactory.initialize(AwsClientFactories.java:151)
	at org.apache.iceberg.aws.AwsClientFactories.loadClientFactory(AwsClientFactories.java:88)
	at org.apache.iceberg.aws.AwsClientFactories.from(AwsClientFactories.java:61)
	at org.apache.iceberg.aws.glue.GlueCatalog.initialize(GlueCatalog.java:141)
```

<img width="1549" alt="image" src="https://github.com/user-attachments/assets/bd9e2a9c-0d5a-483e-95d8-ac42cd4d2ad5">